kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/CustomType.java

```diff
@@ -0,0 +1,33 @@
+package org.kie.dmn.feel.lang;
+
+import java.util.List;
+
+public interface CustomType extends Type {
+
+    List<Field> fields();
```

Comment:
I realized also the proper name of this should be "Property".
This would also make consistent with the annotation-name-change you proposed later, below.
Implemented with Interface + Impl as suggested.
Implemented as Map<String, Property> as suggested.